### DISCUSSION TITLE
compound state: transition and naming isse

### DISCUSSION THREAD

**Discussion by tfahringer** [alexzhornyak/SCXML-tutorial#29] [Q&A]
Let's assume we have a single scxml session/element with 2 compound states in sequence. Can these 2 compound states have child states with identical names but entirely different content/semantics?

Is there any rule in SCXML that prevents using the same name of states across arbitrary nesting of parallel and compound states?

Let's assume there is a compound state getCC with several child states one of them is child state c1. How can I create a transition from within the same scxml session to c1 if c1 is not the initial state? what is the syntax of it?
           <transition event=... target="???" />

what if there is a deep nesting of states such as getCC/c1/cc1/ccc3 ... where cc1 is a child state of c1 and ccc3 is a child state of cc1.
how can i transition to ccc3 from outside? What is the syntax for the target for such a transition?

**Comment by alexzhornyak** [comment]
In single SCXML all names must be unique within session. It is allowed to be duplicated only in submachines called via invoke element
[More info here](https://www.w3.org/TR/scxml/#IDs)

**Comment by alexzhornyak** [comment]
@tfahringer , I would reccomend to download and test scxml in [ScxmlEditor application](https://github.com/alexzhornyak/ScxmlEditor-Tutorial). It has different scxml static checks and you will be able to execute and test it in live session

https://github.com/alexzhornyak/ScxmlEditor-Tutorial

- **Example of  duplicated ID**

![image](https://user-images.githubusercontent.com/18611095/235151442-f92198b0-f3ab-4c3f-9025-4f6377201215.png)

- **Example of scxml element quick documentaion help**

![image](https://user-images.githubusercontent.com/18611095/235151767-46672818-1b49-4434-b387-957a9f21ab78.png)
